src/pj.rs

```diff
@@ -0,0 +1,175 @@
+		for (node_id, con_addr_opt) in connected_peers {
+			assert_eq!(node_id, scheduled_channel.node);
+			println!("Connected to peer: {:?}", node_id);
+			println!("Connected to peer: {:?}", con_addr_opt);
+		}
+	}
+}
+
+#[derive(Clone, Debug)]
+pub struct PayjoinResponse {
+	funding_psbt: Psbt,
+}
+
+impl PayjoinResponse {
+	pub fn new(funding_psbt: Psbt) -> Self {
+		Self { funding_psbt }
+	}
+}
+
+#[derive(Clone, Debug)]
+pub struct PayjoinRequest {
+	original_psbt: Psbt,
+	queue_sender: mpsc::Sender<PayjoinResponse>,
+}
+
+impl PayjoinRequest {
+	fn new(psbt: Psbt, queue_sender: mpsc::Sender<PayjoinResponse>) -> Self {
+		Self { queue_sender, original_psbt: psbt }
+	}
+
+	pub fn original_psbt(&self) -> Psbt {
+		self.original_psbt.clone()
+	}
+
+	pub fn queue(&self, res: PayjoinResponse) {
+		self.queue_sender.try_send(res).unwrap();
+	}
+}
+
+#[derive(Clone)]
+pub struct PayjoinScheduler {
+	queue_sender: mpsc::Sender<PayjoinRequest>,
+}
+
+impl PayjoinScheduler {
+	pub fn new(queue_sender: mpsc::Sender<PayjoinRequest>) -> Self {
+		Self { queue_sender }
+	}
+
+	pub fn schedule(&self, channel: PayjoinRequest) {
+		match self.queue_sender.try_send(channel) {
+			Ok(_) => println!("Payjoin request scheduled"),
+			Err(e) => println!("Error scheduling payjoin request: {:?}", e),
+		}
+	}
+}
+
+fn make_http_response(s: String) -> Result<hyper::Response<Full<bytes::Bytes>>, hyper::Error> {
+	Ok(hyper::Response::builder().body(Full::new(bytes::Bytes::from(s))).unwrap())
+}
+
+async fn payjoin_handler(
+	http_request: Request<hyper::body::Incoming>, pj_scheduler: Arc<Mutex<PayjoinScheduler>>,
+) -> Result<hyper::Response<Full<bytes::Bytes>>, hyper::Error> {
+	match (http_request.method(), http_request.uri().path()) {
+		(&hyper::Method::POST, "/payjoin") => {
+			let headers = http_request.headers().clone();
+			println!("Received payjoin request with headers: {:?}", headers);
+			let body = http_request.into_body().collect().await?;
+			let body = String::from_utf8(body.to_bytes().to_vec()).unwrap();
+			println!("Received payjoin request with body: {:?}", &body);
+			let psbt = Psbt::from_str(&body).unwrap();
+			let (sender, mut receiver) = mpsc::channel::<PayjoinResponse>(1);
+			pj_scheduler.lock().unwrap().schedule(PayjoinRequest::new(psbt, sender));
+			loop {
+				if let Some(resu) = receiver.recv().await {
+					let res = resu.funding_psbt.to_string();
+					return make_http_response(res.into());
+				}
+			}
+		}
+		_ => make_http_response("404".into()),
+	}
+}
+
+impl Service<Request<IncomingBody>> for PayjoinScheduler {
+	type Response = hyper::Response<Full<bytes::Bytes>>;
+	type Error = hyper::Error;
+	type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send>>;
+
+	fn call(&self, http_request: Request<IncomingBody>) -> Self::Future {
+		let state = Arc::new(Mutex::new(self.clone()));
+		let res = match (http_request.method(), http_request.uri().path()) {
+			_ => payjoin_handler(http_request, state),
+		};
```

Comment:
I think methods are being matched twice here for no reason. Unless you're going to transform a custom error type into an http response, you should probably call payjoin_handler directly without the match